File: aribon/2-2/koukano-mondai.cpp
#include <iostream>
#include <algorithm>

using namespace std;

// コインの金額
const int V[6] = {1,5,10,50,100,500};

// input
int C[6];
int A;

void solve(){
    int ans = 0;

    for(int i=5; i>=0; i--){
        int t = min(A/V[i], C[i]);
        A -= t * V[i];
        ans += t;
    }

    printf("%d\n", ans);
}

int main(int argc, char const *argv[]){
    solve();
    return 0;
}

File: aribon/2-2/koukano_mondai.py
# -*- coding:utf-8 -*-
import math


def solve():
    Cs = list(map(int, input().split()))
    A = int(input())

    V = [1,5,10,50,100,500]  # コインの金額

    # 大きな額の硬貨から順番に使い、A円ちょうどを払う
    ans = 0
    for i in range(5, -1, -1):
        t = min(A//V[i], Cs[i])  # コインiを使う枚数
        A -= t * V[i]
        ans += t

    print(ans)


if __name__ == "__main__":
    solve()
